### PR TITLE
relocate kv samples

### LINKED ISSUE #5303: Move Key Vault samples out of `test` directory, into `samples`
Why are they there?

https://github.com/Azure/azure-sdk-for-cpp/tree/main/sdk/keyvault/azure-security-keyvault-administration/test/samples

https://github.com/Azure/azure-sdk-for-cpp/tree/main/sdk/keyvault/azure-security-keyvault-certificates/test/samples

https://github.com/Azure/azure-sdk-for-cpp/tree/main/sdk/keyvault/azure-security-keyvault-keys/test/samples

https://github.com/Azure/azure-sdk-for-cpp/tree/main/sdk/keyvault/azure-security-keyvault-secrets/test/samples

### PR DESCRIPTION
**Pull request by gearama**
closes #5303 
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/main/CONTRIBUTING.md#pull-requests).

- [ ] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [ ] Doxygen docs
- [ ] Unit tests
- [ ] No unwanted commits/changes
- [ ] Descriptive title/description
  - [ ] PR is single purpose
  - [ ] Related issue listed
- [ ] Comments in source
- [ ] No typos
- [ ] Update changelog
- [ ] Not work-in-progress
- [ ] External references or docs updated
- [ ] Self review of PR done
- [ ] Any breaking changes?

### CODE REVIEW

**Review by LarryOsterman — APPROVED** [review]

File: sdk/keyvault/azure-security-keyvault-keys/samples/sample4_encrypt_decrypt.md [hunk 1]
```diff
@@ -72,6 +72,4 @@ std::cout << " - Decrypted data using the algorithm " << decryptResult.Algorithm
 
 ## Source
 
-To see the full example source, see:
-
-- sample4_encrypt_decrypt.cpp
+[defaultazurecredential]: https://github.com/Azure/azure-sdk-for-cpp/blob/main/sdk/identity/azure-identity/README.md
```
Comment (ahsonkhan) [inline sdk/keyvault/azure-security-keyvault-keys/samples/sample4_encrypt_decrypt.md:75]: This link doesn't belong here and doesn't do anything. Remove.

File: sdk/keyvault/azure-security-keyvault-keys/samples/sample4_encrypt_decrypt.md [hunk 2]
```diff
@@ -72,6 +72,4 @@ std::cout << " - Decrypted data using the algorithm " << decryptResult.Algorithm
 
 ## Source
 
-To see the full example source, see:
```
Comment (ahsonkhan) [inline sdk/keyvault/azure-security-keyvault-keys/samples/sample4_encrypt_decrypt.md:75]: Why did we remove this content? Now this heading is empty.

File: sdk/keyvault/azure-security-keyvault-keys/samples/sample5_sign_verify.md [hunk 3]
```diff
@@ -138,6 +138,4 @@ std::cout << " - Verified the signature using the algorithm "
 
 ## Source
 
-To see the full example source, see:
```
Comment (ahsonkhan) [inline sdk/keyvault/azure-security-keyvault-keys/samples/sample5_sign_verify.md:141]: Same here, please revert this change.

File: sdk/keyvault/azure-security-keyvault-certificates/samples/certificate_import_certificate.md [hunk 4]
```diff
@@ -90,4 +90,6 @@ certificateClient.PurgeDeletedCertificate(pemName);
 ## Source
 
 To see the full example source, see:
-[Source Code](https://github.com/Azure/azure-sdk-for-cpp/tree/main/sdk/keyvault/azure-security-keyvault-certificates/test/samples/certificate-import-certificate)
+[Source Code](https://github.com/Azure/azure-sdk-for-cpp/tree/main/sdk/keyvault/azure-security-keyvault-certificates/samples/certificate-import-certificate)
+
+[defaultazurecredential]: https://github.com/Azure/azure-sdk-for-cpp/blob/main/sdk/identity/azure-identity/README.md
```
Comment (ahsonkhan) [inline sdk/keyvault/azure-security-keyvault-certificates/samples/certificate_import_certificate.md:95]: Remove this link. Will stop mentioning

**Review by ahsonkhan — COMMENTED** [review]
This move of samples has some unintended changes like removal of test in md files that we should revert/keep and remove unnecessary links to the identity readme.
